src/ngSanitize/sanitize.js

```diff
@@ -381,12 +387,12 @@ function $SanitizeProvider() {
         if (node.nodeType === 1) {
           handler.end(node.nodeName.toLowerCase());
         }
-        nextNode = node.nextSibling;
+        nextNode = getNonDescendant('nextSibling', node);
         if (!nextNode) {
           while (nextNode == null) {
-            node = node.parentNode;
+            node = getNonDescendant('parentNode', node);
             if (node === inertBodyElement) break;
-            nextNode = node.nextSibling;
+            nextNode = getNonDescendant('nextSibling', node);
             if (node.nodeType === 1) {
```

Comment:
> If an infinite loop is the only concern, disregard my comments.

I meant that is what we are address in this PR :smiley:
But afaict, we are only sanitizing HTML from `ngBindHtml`'s watch-action. Throwing inside a watch-action, will catch the error, pass it to the `$exceptionHandler` and move on to the next watcher.

But if someone uses it differently, it is possible that it breaks the app (as any thrown exception could). Do you think `$sanitize` should be handling its own errors, @koto ?